### DISCUSSION TITLE
Why this error is solved after I reduced the value of n_max?

### DISCUSSION THREAD

**Discussion by Nishijannat** [AstroBridge/BDLensing#58] [Q&A] [answered]
I ran this [code](https://colab.research.google.com/drive/1HTWS2CCkxWwjjb_ZRiMCzZuwHkvTalsN?usp=sharing) and an error showed up which can be seen at the last of this code. There is an indication that the error is somehow related to the shapelet parameter values, so i randomly reduced the n_max value and the problem solved.

**Comment by ajshajib** [comment, marked as answer]
Good catch. The error happens when the chosen `n_max` value is not an integer, which is the strict expectation. So, this line should be like:
```
fixed_source.append({'n_max': 6})
```
